### PR TITLE
[4.0] toggled menu flyout

### PR DESCRIPTION
**Pull request by brianteeman**
Demo of a flyout menu as requested #27347

Now complete with accessible names for the collapsed menu items
(not added a visible text for the toggle bar just an accessible one)

### Testing Instructions
npm i

menu

### CODE REVIEW

**Comment by richard67** [comment]
Drone mysql system test failure seems to be the usual timeout problem and so not related to this PR.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 0a1abf6cba29a917b94292a9c10cda16fa18b989<br /><br />Nice.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27352">issues.joomla.org/tracker/joomla-cms/27352</a>.</sub>

File: administrator/templates/atum/scss/blocks/_sidebar.scss [hunk 1]
```diff
@@ -456,6 +455,13 @@
     display: none;
   }
 
+  .main-nav a:hover .sidebar-item-title {
+    display: inline-block;
+    min-width: 16rem;
+    background-color: var(--atum-bg-dark-80);
+    }
```
Comment (Quy) [inline administrator/templates/atum/scss/blocks/_sidebar.scss]:
```suggestion
  }
```

**Comment by Quy** [comment]
Home Dashboard is behind the buttons and no tooltip for Toggle Menu.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 692cc279889e917ffb3f0f6555396ff6a763c919<br /><br />Even nicer now.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27352">issues.joomla.org/tracker/joomla-cms/27352</a>.</sub>

**Comment by Quy** [comment]
The `Toggle Menu` icon does not have a tooltip. Is it necessary?

**Comment by brianteeman** [comment]
@quy this was only a demo as it still doesn't address the accessibility issue when the sidebar is collapsed. Now it has had positive feedback I will complete the a11y part

**Comment by richard67** [comment]
Ok, I see, I was too fast .. will test again when ready and I have time.

**Comment by brianteeman** [comment]
Now updated with accessibility fixes

**Comment by infograf768** [comment]
`Toggle Menu` still not expanding.

**Comment by brianteeman** [comment]
Not sure what you mean but 
> (not added a visible text for the toggle bar just an accessible one)

**Comment by infograf768** [comment]
toggle

**Comment by brianteeman** [comment]
That's exactly what I wrote and is intentional

**Comment by ChrisHoefliger** [comment]
Also tested successfully. Thanks, Brian.

**Comment by richard67** [comment]
@ChrisHoefliger Could you mark your test result on the issue tracker here [https://issues.joomla.org/tracker/joomla-cms/27352](https://issues.joomla.org/tracker/joomla-cms/27352)?

Just use the "Test this" button and then select the right test result and then submitt.

Thanks in advance.

**Comment by ChrisHoefliger** [comment]
I have tested this item :white_check_mark: successfully on 04f12c0f429cb95331757249dad90ea0ace95c24<br /><br />Successfully tested<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27352">issues.joomla.org/tracker/joomla-cms/27352</a>.</sub>

**Comment by infograf768** [comment]
> That's exactly what I wrote and is intentional

What is the reasoning behind this intention?

**Comment by brianteeman** [comment]
because it is not needed. Not everything requires visible text. what is your reason for saying it should be there

**Comment by infograf768** [comment]
> what is your reason for saying it should be there

Just a normalization with the other icons.

**Comment by brianteeman** [comment]
except it is not part of the menu

**Comment by infograf768** [comment]
Whatever...

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 04f12c0f429cb95331757249dad90ea0ace95c24<br /><br />Not sure if the toggle button should also have some flyout or not. At the one hand it doesn't belong to the menu, at the other hand it looks as if it would. At the one hand it is more or less clear that it must be a toggle button because it has the right icon for that and doesn't have a flyout, at the other hand who knows how other users think.

Finally I think the PR is good as it is, and future enhancement of the toggle button can be done with a new PR.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27352">issues.joomla.org/tracker/joomla-cms/27352</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27352">issues.joomla.org/tracker/joomla-cms/27352</a>.</sub>

**Comment by richard67** [comment]
P.S.: Of course I've checked accessibility, too.

**Comment by HLeithner** [comment]
Thanks

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**